Doc/library/inspect.rst

```diff
@@ -348,26 +348,30 @@ attributes:
 
 .. function:: iscoroutinefunction(object)
 
-   Return ``True`` if the object is a :term:`coroutine function`
-   (a function defined with an :keyword:`async def` syntax).
+   Return ``True`` if the object is a :term:`coroutine function` (a function
+   defined with an :keyword:`async def` syntax), a :func:`functools.partial`
+   wrapping a :term:`coroutine function`, or an instance of a class defining
+   an :keyword:`async def` ``__call__``.
```

Comment:
I would add an explicit mention of sync functions marked with `@markcoroutinefunction` here (and in the versionchanged block too).